### ISSUE TITLE
Disabling io-ompio builds broken on master

### ISSUE THREAD

**Issue by awlauria** [open-mpi/ompi#9919]
There seems to be a new dependency in fcoll on ompio.

Configuring with `---enable-mca-no-build=io-ompio,common-ompio,sharedfp,sharedfp,fbtl,fbtl,fcoll,fcoll,fs` will fail to build with:

```
undefined reference to `mca_common_ompio_set_aggregator_props'
../../../ompi/.libs/libmpi_ftw.so: undefined reference to `mca_common_ompio_decode_datatype'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:1484: ompi_info] Error 1
make[1]: *** [Makefile:2658: all-recursive] Error 1
make: *** [Makefile:1477: all-recursive] Error 1
```

I tracked it back to this commit: 1846e737c173e8854f838afd5296ea6d691a182b

Looking at the code I am not sure what the best spot is to move them to, or how to properly refactor them out.

**Comment by awlauria** [comment]
@edgargabriel I was wrong with my configure statement, I updated it.

I get similar results with --disable-io-ompio (though it is not clear to me that is still supported...)

I do notice that the fcoll/base directory is still built, which is causing the issue. Does this need to be explicitly disabled somehow? I would think including fcoll in the line would disable everything underneath, but for some reason it is still built. I tried adding fcoll-base to the configure line without luck. Similar is happening with the fbtl.
